src/ledger/LedgerTxnOfferSQL.cpp

```diff
@@ -800,34 +800,33 @@ class BulkLoadOffersOperation
         std::vector<LedgerEntry> res;
         while (st.got_data())
         {
+            auto pubKey = KeyUtils::fromStrKey<PublicKey>(sellerID);
+
             // Exclude offers where sellerID in LedgerKey doesn't match sellerID
             // in LedgerEntry
-            auto pubKey = KeyUtils::fromStrKey<PublicKey>(sellerID);
             if (mSellerIDsByOfferID[offerID] == pubKey)
             {
```

Comment:
this looks like condition has been inverted, so either: previous version is wrong, current version is wrong or I'm missing something

Reply:
Yes previous version was wrong. #1974 fixed something wrong into something wrong in a different way.... code matches comment now.